src/context/toast.tsx

```diff
@@ -1,10 +1,19 @@
-import React, { useState, useCallback, useEffect } from "react";
+import React, { useState, useCallback, useEffect, ReactNode } from "react";
```

Comment:
nit. ReactNode import is not necessary.
Since react 17 React is always in scope so this can be 
`import { useState, useCallback, useEffect } from "react";`
and you can access the `ReactNode` type using `React.ReactNode` from anywhere.